services/gateway/endpoints/src/main/java/org/eclipse/ditto/services/gateway/endpoints/routes/cloudevents/CloudEventsRoute.java

```diff
@@ -251,28 +260,52 @@ private CloudEvent validateCloudEvent(final CloudEvent cloudEvent, final Request
         final JsonifiableAdaptable jsonifiableAdaptable = ProtocolFactory.jsonifiableAdaptableFromJson(jsonObject);
         final Signal<?> signal = PROTOCOL_ADAPTER.fromAdaptable(jsonifiableAdaptable);
         final Signal<?> signalWithAdjustedHeaders = signal.setDittoHeaders(
-                        signal.getDittoHeaders().toBuilder().putHeaders(adjustedHeaders).build());
+                signal.getDittoHeaders().toBuilder().putHeaders(adjustedHeaders).build());
         return Optional.of(signalWithAdjustedHeaders);
     }
 
     private void ensureDataContentType(@Nullable final String dataContentType,
             final RequestContext ctx,
             final DittoHeaders dittoHeaders) {
 
-        if (dataContentType == null || !mediaTypeJsonWithFallbacks.contains(dataContentType)) {
+        if (!isCorrectDataType(dataContentType)) {
             if (LOGGER.isInfoEnabled()) {
                 LOGGER.withCorrelationId(dittoHeaders)
                         .info("CloudEvent request rejected: unsupported data-content-type: <{}>  request: <{}>",
                                 dataContentType, requestToLogString(ctx.getRequest()));
             }
             throw UnsupportedMediaTypeException
                     .withDetailedInformationBuilder(dataContentType != null ? dataContentType : "none",
-                            mediaTypeJsonWithFallbacks)
+                            cloudEventsConfig.getDataTypes())
                     .dittoHeaders(dittoHeaders)
                     .build();
         }
     }
 
+    /**
+     * Test if the data type is acceptable.
+     * <p>
+     * A missing, empty or malformed data type is not acceptable.
+     *
+     * @param dataContentType The content type to check.
+     * @return {@code true} if the content type is acceptable, {@code false} otherwise.
+     */
+    private boolean isCorrectDataType(@Nullable final String dataContentType) {
+        if (dataContentType == null) {
+            // no content type
+            return false;
+        }
+
+        final var result = MediaType.parse(dataContentType);
```

Comment:
Please use `akka.http.javadsl.model.ContentTypes.parse(String)` instead (whenever possible, use Akka's "javadsl" packages which contain the Java APIs - using the Scala APIs also work but often looks "strange").
The returned `ContentType` has a `mediaType()` accessor.
Please also don't use `var` keyword here - we did not yet adopt this ;)